ignite/distributed/utils.py

```diff
@@ -737,3 +737,12 @@ def download_dataset():
 
     if current_rank == rank:
         barrier()
+
+
+def _rank_not_in_group(group: Optional[Union[Any, List[int]]]) -> bool:
+    """Check if the current process's rank is not in a given group."""
+    if group is None:
+        return False
+    if isinstance(group, list) and all(isinstance(item, int) for item in group):
+        group = new_group(group)
```

Comment:
Can we hide this inside horovod implementation of _do_all_gather / _rank_not_in_group etc ?